integration-tests/endToEndFlow.spec.ts

```diff
@@ -0,0 +1,151 @@
+import axios from 'axios'
+import { parse } from 'node-html-parser'
+import * as CSV from 'csv-string'
+import {
+  AUDIT_BUCKET_NAME,
+  END_TO_END_TEST_DATE_PREFIX,
+  END_TO_END_TEST_EVENT_ID,
+  END_TO_END_TEST_FILE_NAME
+} from './constants/awsParameters'
+import {
+  endToEndTestRequestDataNoMatch,
+  endToEndTestRequestDataWithMatch
+} from './constants/requestData'
+import { retrieveSecureDownloadDbRecord } from './utils/aws/retrieveSecureDownloadDbRecord'
+import { copyAuditDataFromTestDataBucket } from './utils/aws/s3CopyAuditDataFromTestDataBucket'
+import { deleteAuditDataWithPrefix } from './utils/aws/s3DeleteAuditDataWithPrefix'
+import { getEnv, pause } from './utils/helpers'
+import { approveZendeskTicket } from './utils/zendesk/approveZendeskTicket'
+import { createZendeskTicket } from './utils/zendesk/createZendeskTicket'
+
+describe('Query results generated', () => {
+  jest.setTimeout(60000)
+
+  const getSecureDownloadPageHTML = async (
+    downloadHash: string
+  ): Promise<string> => {
+    try {
+      const response = await axios({
+        url: `${getEnv('QUERY_RESULTS_SECURE_DOWNLOAD_URL')}/${downloadHash}`,
+        method: 'POST'
+      })
+      return response.data
+    } catch (error) {
+      console.log(error)
+      throw 'Could not load secure download page'
+    }
+  }
+
+  const retrieveS3LinkFromHtml = (htmlBody: string): string => {
+    const htmlRoot = parse(htmlBody)
+    const metaTag = htmlRoot.querySelector('meta[http-equiv="refresh"]')
+    const contentAttribute = metaTag?.attributes['content'] as string
+    expect(contentAttribute).toBeDefined()
+
+    const urlMatch = contentAttribute.match(/url=(.*)/)
+    const url = urlMatch ? urlMatch[1] : undefined
+    expect(url).toBeDefined()
+    console.log('S3 URL: ' + url)
+    return url as string
+  }
+
+  const downloadResultsCSVFromLink = async (
+    s3Link: string
+  ): Promise<string> => {
+    try {
+      const response = await axios({ url: s3Link, method: 'GET' })
+      return response.data
+    } catch (error) {
+      throw 'Error downloading results csv from S3 link'
+    }
+  }
+
+  const waitForDownloadHash = async (zendeskId: string): Promise<string> => {
+    let downloadHash = await retrieveSecureDownloadDbRecord(zendeskId)
+    const maxAttempts = 30
```

Comment:
why did you choose 30 max attempts?

Reply:
Well, that was how many attempts I felt would be reasonable based on the time it took for the data to appear during manual testing, and the overall timeout I think is reasonable for the test.

Reply:
educated decision, got ya !